### PR TITLE
337 stopwords n

### PR DESCRIPTION
**Pull request by thanasions**
pipeline bug:
we were doing:
tfidf
*unbias ngrams
reduce dict
apply filyters
unbias ngrams

------
problem is, the first unbias ngrams is wrong. That would mean most bigrams would vanish as all dict is there. The whole point of this technique is to shortlist n features, then hope that redundant high ngrams have gone and the meaningful ones stay so the unbias method works better!

### CODE REVIEW

File: tests/test_nmf_wrapper.py [hunk 1]
```diff
@@ -21,31 +21,31 @@ def setUpClass(cls):
         cls.__nmf = nmf_topic_modelling(nmf_topics, tfidf_obj.tfidf_matrix)
 
     def test_nmf_topic1(self):
```
Comment (IanGrimstead) [inline tests/test_nmf_wrapper.py]: Scores are much lower by an order of magnitude - is that ok?

File: tests/test_terms_graph.py [hunk 2]
```diff
@@ -56,34 +56,40 @@ def test_num_nodes(self):
         self.assertEquals(50, len(self.__nodes))
 
     def test_num_links(self):
-        self.assertEquals(454, len(self.__links))
+        self.assertEquals(122, len(self.__links))
 
     def test_terms_in_nodes(self):
         texts = [x['text'] for x in self.__nodes]
```
Comment (IanGrimstead) [inline tests/test_terms_graph.py]: Any reason for changes in n-gram selection?

File: tests/test_terms_graph.py [hunk 3]
```diff
@@ -56,34 +56,40 @@ def test_num_nodes(self):
         self.assertEquals(50, len(self.__nodes))
 
     def test_num_links(self):
-        self.assertEquals(454, len(self.__links))
+        self.assertEquals(122, len(self.__links))
```
Comment (IanGrimstead) [inline tests/test_terms_graph.py]: That's a big change - any reason?

File: tests/test_nmf_wrapper.py (hunk 1 above)
Comment (thanasions) [reply tests/test_nmf_wrapper.py]: random?

File: tests/test_tfidf_reduce.py [hunk 4]
```diff
@@ -12,6 +12,7 @@ class TestTfidfReduce(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
+        cls.maxDiff = None
```
[the remark this answers is not in the document]
Comment (thanasions) [reply tests/test_tfidf_reduce.py]: tells the test engine to allow larger diff files to be returned should an error occur...

**Comment by IanGrimstead** [comment]
Well I never! New feature to me, nice one 😊

**Pull request closed without merging**